tests/python/test_usdt.py

```diff
@@ -19,25 +20,44 @@ def setUp(self):
         # Application, minimum, to define three trace points
         app_text = b"""
 #include <unistd.h>
-#include <folly/tracing/StaticTracepoint.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include "folly/tracing/StaticTracepoint.h"
 
 int main() {
   char s[100];
   int i, a = 200, b = 40;
   for (i = 0; i < 100; i++) s[i] = (i & 7) + (i & 6);
+  uint64_t j = 0;
+  char s1[64];
+  const char* str = "str";
+  size_t len = strlen(str);
   while (1) {
     FOLLY_SDT(test, probe_point_1, s[7], b);
     FOLLY_SDT(test, probe_point_3, a, b);
     FOLLY_SDT(test, probe_point_1, s[4], a);
     FOLLY_SDT(test, probe_point_2, 5, s[10]);
     FOLLY_SDT(test, probe_point_3, s[4], s[7]);
+
+    memset(&s1, '\0', sizeof(s1));
+    strncpy(s1, str, len);
+    snprintf(s1 + len, sizeof(s1) - len, "%d", j);
+    FOLLY_SDT(test, probe_point_4, j++, &s1);
```

Comment:
Actually, these questions would have been more appropriate in the context of lib1.cpp.

I ran into some shenanigans when I provided the FOLLY_SDT macro with a hardcoded string, such as:
    FOLLY_SDT(test, probe_point_4, "bla123456", j++);
BPF comes back with a compilation error:
/virtual/main.c:55:21: error: use of undeclared identifier 'int80_t'
  if (len != sizeof(int80_t)) return -1;

Note that the erroneous identifier depends on the length of the hardcoded string, so if you specify "bla", it compiles (int32_t). :)

One can avoid this by creating a temporary const char* variable.

However, the above was reason for me to panic :) and start copying strings to a known-size stack allocated buffer before passing it along to BPF. That's a bit unnecessary, so I updated lib1.cpp, see my next commit. All's good now.